clippy_lints/src/loops/single_element_loop.rs

```diff
@@ -87,14 +87,27 @@ pub(super) fn check<'tcx>(
             arg_snip = format!("({arg_snip})").into();
         }
 
-        span_lint_and_sugg(
-            cx,
-            SINGLE_ELEMENT_LOOP,
-            expr.span,
-            "for loop over a single element",
-            "try",
-            format!("{{\n{indent}let {pat_snip} = {prefix}{arg_snip};{block_str}}}"),
-            applicability,
-        );
+        if clippy_utils::higher::Range::hir(arg_expression).is_some() {
+            let sugg = snippet(cx, arg_expression.span, "..");
+            span_lint_and_sugg(
+                cx,
+                SINGLE_ELEMENT_LOOP,
+                arg.span,
+                "for loop over a single range inside an array, rather than iterating over the elements in the range directly",
```

Comment:
Perhaps we can find a shorter way to phrase this? I like the clarity though. Maybe something like "This loops only once with <loop pattern> being <range expr>", which is admittedly harder to construct.

Reply:
We discussed a lot about the message of this lint, but opted for a longer and more verbose one. I adjusted it to contain your version, which is shorter, but now the text "0..5" is written twice in the lint message, which feels redundant to me.

Reply:
I see it as "item" (the looping variable name) and "0..5", rather than 0..5 twice. Am I missing something?

Reply:
This is the current output of the lint:
```
error: this loops only once with item being 0..5
  --> $DIR/single_element_loop.rs:24:17
   |
LL |     for item in [0..5] {
   |                 ^^^^^^ help: did you mean to iterate over the range instead?: `0..5`
```

"0-5" is mentioned in the lint error message and in the suggestion. I thing mentioning it once should be enough, but mentioning it twice will probably not hurt.

Reply:
... and with a lowercase "t" the error message actually passes the tests, which were failing with my commit from earlier today. :)